7a_temp_coop_munge/src/data_parsers/parse_UniversityofMissouri_LimnoProfiles_2017_2020.R

```diff
@@ -0,0 +1,203 @@
+  groupings <- c(files_2017_hw, files_2017_092_hw, files_2018_hw,
+                 files_2017, files_2018_2020)
+  if ((length(files_from_zip) - 1) != lapply(groupings, length) %>% unlist %>% sum) {
+    stop('Number of files in subfolders does not match grouping')
+  }
+
+  # column selection for each data grouping
+  # column names will be forced `tolower` in the parsers because
+  # case is inconsistent file-to-file
+  keep_cols_2017 <- c('Date', 'Time', 'Dep.M', 'X.C')
+  keep_cols_2018_2020 <- c('Date..MM.DD.YYYY.', 'Time..HH.MM.SS.',
+                           'Depth.m', 'Temp..C')
+
+  # Clean data from 2017 with consistent formatting --------------------
+  dat_2017 <- files_2017 %>%
+    purrr::map_dfr(~ read_subset_consistent_data(., keep_cols = keep_cols_2017)) %>%
+    dplyr::mutate(
+      DateTime = lubridate::mdy(date),
+      time = strftime(time, "%H:%M"),
+      Timezone = NA,
+      depth = dep.m,
+      temp = x.c,
+      Missouri_ID = lake_id,
+      .keep = 'none'
+    )
+
+  # Clean data from 2018-2020 with consistent formatting --------------------
+  dat_2018_2020 <- files_2018_2020 %>%
+    purrr::map_dfr(~ read_subset_consistent_data(., keep_cols = keep_cols_2018_2020)) %>%
+    dplyr::mutate(
+      DateTime = lubridate::mdy(date..mm.dd.yyyy.),
+      time = strftime(time..hh.mm.ss., "%H:%M"),
+      Timezone = NA,
+      depth = depth.m,
+      temp = temp..c,
+      Missouri_ID = lake_id,
+      .keep = 'none'
+    )
+
+  # Clean 2017 & 2018 data from hand-transcribed field sheets ----------------------
+
+  dat_2017_hw <- files_2017_hw %>%
+    purrr::map_dfr(~ read_subset_hw_files(.))
+
+  dat_2017_hw_092 <- read_subset_hw_files(files_2017_092_hw,
+                                          skip_rows = 15,
+                                          skip_cols = 2,
+                                          depth_position = 16,
+                                          temp_position = 3)
+
+  dat_2018_hw <- files_2018_hw %>%
+    purrr::map_dfr(~ read_subset_hw_files(., skip_rows = 0))
+
+  all_hw_files <- dplyr::bind_rows(dat_2017_hw, dat_2017_hw_092, dat_2018_hw) %>%
+    dplyr::mutate(
+      DateTime = date,
+      time = time,
+      Timezone = NA,
+      depth = depth,
+      temp = temp,
+      Missouri_ID = lake_id,
+      .keep = 'none'
+    )
+
+  # combine all data sets together ------------------------------------------
+  data_clean <- dplyr::bind_rows(
+    dat_2017,
+    dat_2018_2020,
+    all_hw_files
+  )
+
+  saveRDS(object = data_clean, file = outfile)
+  sc_indicate(ind_file = outind, data_file = outfile)
+
+}
+
+# Read data functions -----------------------------------
+
+#' Read and clean files from 2019-2020 UMO data
+#'
+#' @param full_path chr, full file path
+#' @param keep_cols chr, vector of column names to keep
+#'
+read_subset_consistent_data <- function(full_path, keep_cols) {
+  dat <- suppressMessages(readr::read_csv(full_path, name_repair = 'universal') )
```

Comment:
I needed to change this to 
```r
  dat <- suppressMessages(readr::read_csv(full_path, name_repair = 'universal', locale = locale(encoding = "latin1")))
```
To make this compatible with the mac build

Reply:
Again! I'll add it in and test